frontends/election-manager/src/screens/tally_screen.tsx

```diff
@@ -152,7 +154,11 @@ export function TallyScreen(): JSX.Element {
               Load CVR Files
             </Button>{' '}
             <Button
-              disabled={!castVoteRecordFiles.wereAdded || isOfficialResults}
+              disabled={
+                cvrFilesQuery.isFetching ||
```

Comment:
Is there a reason you're using `isFetching` instead of `isLoading`?

Reply:
I definitely agree we should consider this state.

What do you think about using the approach we discussed earlier? Handle `isLoading` at the top-level of the component for the initial fetch and then make sure that mutations that invalidate the cache wait for the `queryClient.invalidateQueries` promise to resolve.

Reply:
i worry about a component making too many assumptions about how and when data ends up in certain states (if that makes sense). if we only handle `isLoading`, we miss the case where the query's in `isFetching` state,  and the data after the fetch could be different from what was in the cache before. from our vantage point, we know it's unlikely because of the way the product works, but not handling that state makes the component less resilient against some potential future change down the line.

so, less concerned about this specific case and more so about adopting a consistent pattern that helps us avoid unexpected edge cases, tough-to-diagnose bugs, or even just weird UI flicker.

agreed that we should go through and do a sweeping change to make all our mutations wait on invalidations though, but that's just one of a few ways for this to end up in the `isFetching` state, since just navigating back and forth will do that

Reply:
Yeah, I think I see what you're getting at. I agree it seems like something we should figure out a consistent pattern for. Maybe we can discuss sync tomorrow?

It would be helpful to have a list of all the cases we need to consider. So far, here's what I think we've surfaced:
- Initial load when component mounts
- Refetch after invalidation from a mutation
- Navigating to a diff screen and coming back (q: in what ways is this different from initial load?)

Are there others you have in mind? I know there are also the refetchOnWindowFocus and refetchOnReconnect params.

Some relevant docs I've been reading to help think about this (which you may have already seen):
- https://tanstack.com/query/v4/docs/guides/queries#why-two-different-states
- https://tanstack.com/query/v4/docs/guides/background-fetching-indicators

Reply:
yeah, sync tomorrow sounds good! maybe we can stay on after standup and anyone else interested can linger?
i'll give the different cases some thought ahead of time - thanks!